api/src/main/java/edu/cornell/mannlib/vitro/webapp/dynapi/computation/ExecutionTree.java

```diff
@@ -0,0 +1,98 @@
+package edu.cornell.mannlib.vitro.webapp.dynapi.computation;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+public class ExecutionTree {
+    private Map<StepInfo, List<StepInfo>> forwardMap;
+    private Map<StepInfo, List<StepInfo>> backwardMap;
+    
+    public ExecutionTree() {
```

Comment:
Should this be private to prevent instantiation of uninitialized execution tree.

Reply:
Fixed